### ISSUE TITLE
Fullscreen viewer generating a logic error

### ISSUE THREAD

**Issue by granitepenguin** [wwivbbs/wwiv#1111]
WWIV: 5.3.2680
OS: Linux - debian 8

I'm in my WWIVnet General Chat that currently has 3155 messages.  When I try to (S)can messages, or scroll down, I'm getting the following message and getting dumped back to the Main menu

Caught std::logic_error: basic_string::_S_create

It seems to be tied to anything trying to display info about message 3151.  I can PgDn or scroll down just fine until the summary line for 3151 is displayed.

If I'm actually in the full messages, it displays the message contents just fine.  It's only in the FS summary where it drops.

I'm also seeing it in WWIV Development (1808 messages), but not in WWIVnet sysop (1881 messages), so it doesn't seem to be directly related to sub size.

**Comment by granitepenguin** [comment]
a catch catch doesn't seem to be terribly helpful, but here it is

```
Catchpoint 1 (exception caught), 0x00007ffff72b2d11 in __cxa_begin_catch ()
        from /usr/lib/x86_64-linux-gnu/libstdc++.so.6                                
(gdb) bt                                                                        
#0  0x00007ffff72b2d11 in __cxa_begin_catch ()                                     
from /usr/lib/x86_64-linux-gnu/libstdc++.so.6                                
#1  0x000000000063f778 in Application::Run(int, char**) ()                      
#2  0x00000000006215f4 in bbsmain(int, char**) ()                               
#3  0x0000000000620fc6 in main ()                                               
(gdb) li                                                                        
1       dl-debug.c: No such file or directory.                                  
(gdb) cont                                                                      Continuing.
Caught std::logic_error: basic_string::_S_create
```
I'll try to get a better look pre-fail

**Comment by granitepenguin** [comment]
possibly some better info:

```
Catchpoint 1 (exception thrown), 0x00007ffff72b3dbd in __cxa_throw ()
              from /usr/lib/x86_64-linux-gnu/libstdc++.so.6                                
(gdb) bt
#0  0x00007ffff72b3dbd in __cxa_throw ()
   from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#1  0x00007ffff7309ac6 in std::__throw_length_error(char const*) ()
   from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#2  0x00007ffff7313eb2 in std::string::_Rep::_S_create(unsigned long, unsigned long, std::allocator<char> const&) ()
   from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#3  0x00007ffff73149a1 in ?? () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#4  0x00007ffff7314a4b in std::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(unsigned long, char, std::allocator<char> const&) () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#5  0x00000000006d58d8 in CreateLine(std::unique_ptr<wwiv::sdk::msgapi::Message, std::default_delete<wwiv::sdk::msgapi::Message> >&&, int) ()
#6  0x00000000006d5bf4 in CreateMessageTitleVector(wwiv::sdk::msgapi::MessageArea*, int, int) ()
#7  0x00000000006d621f in HandleListTitlesFullScreen(int&, MsgScanOption&) ()
#8  0x00000000006d9883 in scan_new(int, MsgScanOption, bool&, bool) ()
#9  0x00000000006d9d7a in scan(int, MsgScanOption, bool&, bool) ()
#10 0x00000000006d05f7 in ScanMessageTitles() ()
#11 0x00000000006c407b in TitleScan() ()
#12 0x000000000069ce05 in wwiv::menus::CreateCommandMap()::{lambda(wwiv::menus::MenuItemContext&)#50}::operator()(wwiv::menus::MenuItemContext&) const ()
#13 0x00000000006ae3fc in std::_Function_handler<void (wwiv::menus::MenuItemContext&), wwiv::menus::CreateCommandMap()::{lambda(wwiv::menus::MenuItemContext&)#50}>::_M_invoke(std::_Any_data const&, wwiv::menus::MenuItemContext&) ()
#14 0x00000000006be203 in std::function<void (wwiv::menus::MenuItemContext&)>::operator()(wwiv::menus::MenuItemContext&) const ()
#15 0x000000000069c3b1 in wwiv::menus::InterpretCommand(wwiv::menus::MenuInstance*, std::string const&) ()
#16 0x000000000069344c in wwiv::menus::MenuInstance::MenuExecuteCommand(std::string const&) ()
#17 0x0000000000691ec2 in wwiv::menus::MenuInstance::RunMenu() ()
#18 0x00000000006914cc in wwiv::menus::StartMenus() ()
#19 0x0000000000691d42 in wwiv::menus::mainmenu() ()
#20 0x000000000063f1d7 in Application::Run(int, char**) ()
#21 0x00000000006215f4 in bbsmain(int, char**) ()
#22 0x0000000000620fc6 in main ()
Continuing.
Caught std::logic_error: basic_string::_S_create
```

**Comment by wwiv** [comment]
I think this might be fixed in today's build. I tried to fix it but can't reproduce it.

**Comment by wwiv** [comment]
Wasn't fixed before, fixed some other crash.  
Fixed in 291479bd13bfdf73c507614646f815d76d2665d3

**Comment by granitepenguin** [comment]
So far, so good with 2684
